### DISCUSSION TITLE
Filter out small dimension images

### DISCUSSION THREAD

**Discussion by phedlund** [onevcat/Kingfisher#1824] [Q&A]
Hi,

I extract image urls from html and use `ImagePrefetcher` to download the images. What would be the best way to filter out images smaller than say 100x100px?

**Comment by onevcat** [comment]
`ImagePrefetcher` will only give you a list of skipped/downloaded images source, instead of giving you the actual images. Considering the fact that the loaded resource will be cached by default, maybe you can try:

```swift
ImagePrefetcher(urls: urls) { skippedResources, failedResources, completedResources in
  let smallImagesInMemoryCache = (skippedResources + completedResources).compactMap { r in
    ImageCache.default.retrieveImageInMemoryCache(forKey: r.cacheKey)
  }.filter { image in
    image.size.width < 100 && image.size.height < 100
  }
}
```

However, be caution that if you have lots of images, it might be possible that the downloaded images are removed from the memory cache but only stored in the disk cache. In this situation, you have to also use the disk cache related method to check them.
